### ISSUE TITLE
Calendar/Date Picker Component

### ISSUE THREAD

**Issue by bilal-23** [bilal-23/no-js#7]
Implement a date selection component with month/year navigation. Should be keyboard accessible, support date ranges, and work without external libraries.
